irodsConnector/tickets.py

```diff
@@ -43,41 +43,13 @@ def create_ticket(self, obj_path: str, expiry_string: str = '') -> tuple:
             (str, bool)
 
         """
-        ticket_id = ''.join(choice(ascii_letters) for _ in range(20))
-        ticket = irods.ticket.Ticket(self.sess_man.irods_session, ticket_id)
+        ticket = irods.ticket.Ticket(self.sess_man.irods_session)
         ticket.issue('read', obj_path)
         logging.info('CREATE TICKET: %s: %s', ticket.ticket, obj_path)
         expiration_set = False
         if expiry_string != '':
             try:
-                expiration_set = self._modify_ticket(ticket, expiry_string)
+                expiration_set = ticket.modify('expire', expiry_string) == ticket
```

Comment:
Do you know if this bug is fixed now?

Reply:
I have no easy way to test it fully.  I expect somebody else can.  The limit if my testing is sanity checks such as finding no syntax errors.  Do we have any kind of unit tests for these modules?

Reply:
That is true. With the python API one could not set an expiry date and one had no function to clean up old tickets (not as user and not as admin). So we would still need the icommands here.

Reply:
Is there a python-irodsclient issue noting this lack of functionality?   A quick search didn't reveal one...

/cc @d-w-moore

Reply:
I have tested the PRC functionality to create and modify a Ticket with the expiration date using the same methods in the connector and they all work fine.  I have not seen any functionality in iBridges for clearing tickets, but the iRODS Ticket() class does provide a working `delete()` method.  Unfortunately, querying for tickets is a bit round-about, but it does work.  What other functionality is needed?

When I tried all this using the iBridges ticket login, I keep getting segmentation faults when I change focus under macOS.  I'm trying under Linux, but I'm having to solve what looks like minor bugs.

Shall we make adding PRC query and delete functionality to this Issue/PR or should we create a new one?  I any case, I would like to solve the bugs I have just seen before this PR is ready to merge.

Reply:
I have just tested the functionality through the GUI on Mac. So the ticket was created with the Python API and an expiry date was successfully set. I confirmed that through the icommands. Issue is solved.

The deletion of tickets according to their creation date or expiry date, has not been implemented yet in our backend. I will pout it in the issue list. I can remember that there was an issue with table names and how the table was accessed through the c++ API?! @trel

Reply:
@jpmcfarland I also hadn't noted (yet) any such bugs with the PRC ticket functionality.  Tickets may also be queried and deleted (examples are in the tests), but I certainly wouldn't deny the possibility of unknown bugs and unexpected behaviors.

Reply:
It turns out the only bugs I'm currently seeing are related to a module requiring updates after the recent set of major refactoring.  I believe this was simply a low priority task and I stumbled into it by noticing an unusual inconsistency.